src/gamma.jl

```diff
@@ -754,8 +754,13 @@ end
 @inline lgamma(x::Real) = lgamma(float(x))
 
 ## from base/numbers.jl
-# TODO: deprecate instead of doing this type-piracy here?
-Base.factorial(x::Number) = gamma(x + 1) # fallback for x not Integer
+
+# this trickery is needed while the deprecated method in Base exists
+@static if !hasmethod(Base.factorial, Tuple{Number})
+    import Base: factorial
+end
+factorial(x) = Base.factorial(x) # to make SpecialFunctions.factorial work unconditionally
```

Comment:
We should probably narrow the signature here to match Base. `x::Integer` would probably be sufficient, though Base also appears to include a method for `BigFloat`.

Reply:
Does it really matter if the `MethodError` is thrown here or from `Base.factorial`?